src/notarytool.ts

```diff
@@ -83,24 +83,45 @@ export async function notarizeAndWaitForNotaryTool(opts: NotaryToolStartOptions)
     ];
 
     const result = await spawn('xcrun', notarizeArgs);
-    const parsed = JSON.parse(result.output.trim());
 
-    if (result.code !== 0 || !parsed.status || parsed.status !== 'Accepted') {
+    if (result.code !== 0) {
+      let parsed: any;
       try {
-        if (parsed && parsed.id) {
+        parsed = JSON.parse(result.output.trim());
+      } catch (err) {
+        throw new Error(
+          `Failed to notarize via notarytool.  Failed with unexpected result: \n\n${result.output.trim()}`,
+        );
+      }
+
+      let logId: undefined | string;
+      if (!parsed.status || parsed.status !== 'Accepted') {
+        logId = parsed.id;
```

Comment:
I think this is needlessly involved, and it's *especially* important to get `logId` if the status is not `Accepted`.

Simply `const logId = parsed.id` or even just:

```
if (parsed.id) {
  // get the log with parsed.id
}
```

Reply:
Yeah agreed, this PR should address the feedback.